### PR TITLE
Provide a default CopyStrategy overload for copyItem.

### PR DESCRIPTION
**Pull request by UncleMattHope**
FilesystemProtocol/copyItem has an overload providing the default CopyStrategy

### Motivation:

Based on user feedback from #2806 
This makes it easier to use, especially when copying files such that it is irrelevant.

### Modifications:

Add overload providing ```(at:to:shouldProceedAfterError:shouldCopyItem:)``` with ```CopyStrategy.platformDefault```
as the strategy.

### Result:

A simpler and more robust to change API will exist for ```copyItem()```

### CODE REVIEW

**Review by weissi — APPROVED** [review]
Thank you!

File: Sources/NIOFileSystem/FileSystemProtocol.swift [hunk 1]
```diff
@@ -535,6 +535,52 @@ extension FileSystemProtocol {
...
+    ///   - shouldCopyItem: A closure which is executed before each copy to determine whether each
+    ///       item should be copied. See Filtering section for full details
+    ///
+    /// #### Parallelism
+    ///
+    /// This overload uses ``CopyStrategy/platformDefault`` which is likely to result in multiple concurrency domains being used
+    /// in the event of copying a directory.
+    /// See the detailed description on ``copyItem(at:to:strategy:shouldProceedAfterError:shouldCopyItem:)``
+    /// for the implications of this with respect to the `shouldProceedAfterError` and `shouldCopyItem` callbacks
+    func copyItem(
```
Comment (glbrntt) [inline Sources/NIOFileSystem/FileSystemProtocol.swift:565]: Needs a `public`!

**Review by glbrntt — CHANGES_REQUESTED** [review]

**Review by glbrntt — APPROVED** [review]

File: Sources/NIOFileSystem/Internal/ParallelDirCopy.swift [hunk 2]
```diff
@@ -134,6 +134,8 @@ extension FileSystem {
                     if let item = item {
                         keepConsuming = !onNextItem(item)
                     } else {
+                        // To accurately propagate the cancellation we must check here too
```
Comment (UncleMattHope) [inline Sources/NIOFileSystem/Internal/ParallelDirCopy.swift:137]: This fixes the flaky test, and makes cancellation reliable.

File: Tests/NIOFileSystemIntegrationTests/FileSystemTests.swift [hunk 3]
```diff
@@ -791,10 +791,49 @@ final class FileSystemTests: XCTestCase {
         _ copyStrategy: CopyStrategy,
         line: UInt = #line
     ) async throws {
-        let path = try await self.fs.temporaryFilePath()
+        // Whitebox testing to cover specific scenarios
```
Comment (UncleMattHope) [inline Tests/NIOFileSystemIntegrationTests/FileSystemTests.swift:794]: This makes the test non flaky (it reliably fails without the fix)

**Review by glbrntt — APPROVED** [review]

**Pull request merged**